### PR TITLE
[BugFix][v0.23.0] Avoid partial MiniMax parameter arguments

### LINKED ISSUE #10758: [Bug]: [v0.20.2rc1]minimax-m2.5工具调用末尾会拼接</parameter>导致路径识别失败
### Your current environment
Atlas 800I A2 
HDK 25.5.1
vllm ascend v0.20.2rc1

<details>
<summary>The output of `python collect_env.py`</summary>

```text
```

</details>
### 🐛 Describe the bug

通过bad case 抓取几个日志：rendered.json、rendered_prompt.json、raw_completion.json

[bad_case.json](https://github.com/user-attachments/files/29189406/bad_case.json)

[raw_completion.json](https://github.com/user-attachments/files/29189410/raw_completion.json)

[rendered.json](https://github.com/user-attachments/files/29189411/rendered.json)

[rendered_prompt.json](https://github.com/user-attachments/files/29189412/rendered_prompt.json)

Here's detailed commands we used to start service:
【minimax-m2.5-w8a8-QuaRot】：
export LD_PRELOAD=/usr/lib64/libjemalloc.so.2:$LD_PRELOAD
export HCCL_OP_EXPANSION_MODE="AIV"
export HCCL_BUFFSIZE=512

sysctl -w vm.swappiness=0
sysctl -w kernel.numa_balancing=0
sysctl kernel.sched_migration_cost_ns=50000

export TASK_QUEUE_ENABLE=1
export VLLM_ASCEND_ENABLE_FUSED_MC2=1
export PYTORCH_NPU_ALLOC_CONF=expandable_segments:True

export VLLM_ASCEND_ENABLE_FLASHCOMM1=1
export VLLM_ASCEND_BALANCE_SCHEDULING=1
export HCCL_INTRA_PCIE_ENABLE=1
export HCCL_INTRA_ROCE_ENABLE=0

export OMP_PROC_BIND=false
export OMP_NUM_THREADS=8

export PYTHONPATH=/vllm-workspace/vllm:$PYTHONPATH

vllm serve /appdata/model/MiniMax-M2.5-w8a8-QuaRot/ \
    --served-model-name MiniMax-M2.5 \
    --safetensors-load-strategy eager \
    --host 0.0.0.0 \
    --port 8080 \
    --trust-remote-code \
    --tensor-parallel-size 8 \
    --quantization ascend \
    --enable-expert-parallel \
    --enable-chunked-prefill \
    --enable-prefix-caching \
    --max-num-seqs 128 \
    --max-num-batched-tokens 6144 \
    --compilation-config '{"cudagraph_mode": "FULL_DECODE_ONLY","cudagraph_capture_sizes": [1,4,16,32,128,256,400,600,800]}' \
    --gpu-memory-utilization 0.9 \
    --enable-auto-tool-choice \
    --tool-call-parser minimax_m2 \
    --reasoning-parser minimax_m2_append_think \
    --enable-force-include-usage \
    --additional-config '{"enable_cpu_binding":true}' \
    --model-loader-extra-config '{"enable_multithread_load":true,"num_threads":16}' \
    --speculative_config '{"method": "eagle3", "model": "/appdata/model/MiniMax-M2.5-eagel-0322/",  "num_speculative_tokens":3}'

### PR DESCRIPTION
**Pull request by QwertyJack**
### What this PR does / why we need it?

Fixes #10758. Backports the MiniMax streaming fixes from #11384 to `releases/v0.23.0`.

The main-branch #11384 can remove the stale local MiniMax parser patch because main has moved to the newer vLLM parser-engine MiniMax path. The `releases/v0.23.0` branch is pinned to vLLM `v0.23.0`, where that MiniMax parser-engine module is not available, so this release PR keeps the local MiniMax parser patch and ports the two reproduced #10758 fixes into it:

- stream MiniMax-M2 argument deltas only after the current `<parameter>` has a complete `</parameter>` end tag, preventing split tag pieces such as `</`, `parameter`, and `>` from leaking into JSON arguments
- avoid constructing `DeltaMessage(tool_calls=None)` when plain content and a partial tool-call start arrive in the same streaming chunk; the resulting content-only delta serializes without `tool_calls`, matching https://github.com/vllm-project/vllm/issues/44105

This PR also removes the obsolete MiniMax worker-side manual TP all-reduce call while keeping the NPU fp32 router-logits behavior. The verified vLLM MoE runner no longer exposes that manual reduction hook, and leaving it in the release branch blocks MiniMax TP8+EP startup.

This intentionally does not copy #10377's broad empty-list workaround.

### Does this PR introduce _any_ user-facing change?

Yes. Streaming MiniMax-M2 tool-call argument deltas are emitted after each parameter closes instead of while a parameter is still open. This prevents malformed partial XML tag pieces from appearing in streamed JSON arguments. Content-only deltas before a partial tool call also serialize without empty `tool_calls`.

The worker-side change is a compatibility cleanup for MiniMax TP/EP serving on the verified vLLM stack; it removes a stale manual reduction call that the current MoE runner no longer supports.

### How was this patch tested?

- `python -m pytest -q tests/ut/patch/platform/test_patch_minimax_m2_tool_call_parser.py tests/ut/patch/platform/test_patch_minimax_usage_accounting.py`
- Replayed the #10758 raw completion with the MiniMax-M2.5 tokenizer; streamed fragments reconstructed `{"path":"a.txt","content":"123"}` without `</parameter>` leakage.
- Reproduced the leading-content plus partial-tool-call-start path and verified it returns content with an empty internal default while serialized payloads omit `tool_calls`.
- `ruff check vllm_ascend/patch/platform/patch_minimax_m2_tool_call_parser.py tests/ut/patch/platform/test_patch_minimax_m2_tool_call_parser.py`
- `python -m py_compile vllm_ascend/patch/platform/patch_minimax_m2_tool_call_parser.py`
- `ruff check vllm_ascend/patch/worker/patch_minimax_m2.py vllm_ascend/patch/__init__.py`
- `python -m py_compile vllm_ascend/patch/worker/patch_minimax_m2.py vllm_ascend/patch/__init__.py`
- Live MiniMax-M2.5 real-weight service on TP8+EP with the verified vLLM main commit:
  - `/v1/models` returned 200
  - text chat request returned 200 with non-empty output
  - non-streaming auto `write_file` tool request returned 200 with arguments `{"path": "a.txt", "content": "123"}`
  - streaming auto `write_file` tool request returned 200; 30 SSE chunks / 3 argument fragments reconstructed `{"path": "a.txt", "content": "123"}` with no `<parameter>` / `</parameter>` leakage and no `tool_calls: null` deltas

Note: live validation used `xgrammar==0.2.1`, matching the verified vLLM checkout requirement.

- vLLM version: v0.23.0
- vLLM main: https://github.com/vllm-project/vllm/commit/ee0da84ab9e04ac7610e28580af62c365e898389

### CODE REVIEW

File: vllm_ascend/patch/platform/patch_minimax_m2_tool_call_parser.py [hunk 1]
```diff
@@ -297,23 +272,19 @@ def _build_partial_arguments(
             param_value = invoke_body[value_start:]
             search_pos = len(invoke_body)
 
-        if not param_complete and not param_value.strip():
+        if not param_complete:
             break
```
[the remark this answers is not in the document]
Comment (QwertyJack) [reply vllm_ascend/patch/platform/patch_minimax_m2_tool_call_parser.py:269]: Good catch. The incomplete-parameter branch was no longer observable after the early break, so I refactored this block to break first and only assign `param_value`/`search_pos` for complete parameters.

**Review by linfeng-yuan — APPROVED** [review]
